RLBotDiscordBot/bot.py

```diff
@@ -36,17 +38,35 @@ def __init__(self):
         self.remove_command('help')
 
         for extension in initial_extensions:
+            #print("woof")
             try:
                 self.load_extension(extension)
             except Exception as e:
                 self.logger.error(f'There was an error loading the extension {extension}. Error: {e}')
 
     async def on_ready(self):
         self.logger.info(f'{self.user} online! (ID: {self.user.id})')
+        self.has_reacted = 0
+        self.has_checked = False
 
     async def on_message(self, message):
+        if not self.has_checked:
+            x = message.guild.members
+            z = message.guild.roles
+            for member in x:
+                y = member.roles
+                for role in y:
+                    if role == get(member.guild.roles, name="Python") or role == get(member.guild.roles, name="Java") or role == get(member.guild.roles, name="C#") or role == get(member.guild.roles, name="Rust") or role == get(member.guild.roles, name="Scratch"):
+                        new_role = get(member.guild.roles, name="botmaker")
+                        await member.add_roles(new_role, reason=None, atomic=True)
+
         if message.author.bot:
-            return
+            if message.channel.id == 352507627928027138:
+                self.has_reacted += 1
+                if self.has_reacted % 2 == 0:
```

Comment:
Why only react half the time?

Reply:
Because currently, I have it sending 2 messages, one with the info about who sent it and the other with the clip, I set it up so it only reacts on the clip rather than reacting on both